core/spec/models/spree/stock/estimator_spec.rb

```diff
@@ -140,6 +140,32 @@ module Stock
             expect(shipping_rates.first.tax_rate).to eq(tax_rate)
           end
         end
+
+        it 'uses the configured shipping rate selector' do
+          class Spree::Stock::TestSelector; end;
```

Comment:
> it appears the global state gets cleared :)

Great!

> all that matters is the selector and sorter implement find_default and sort

Personally I side with DHH/Fowler/Beck on this (e.g. [this](http://martinfowler.com/articles/mocksArentStubs.html) and [this](https://www.thoughtworks.com/insights/blog/mockists-are-dead-long-live-classicists)) and avoid mocking when it makes sense, and I think it makes sense here.  But I won't hold up this PR if other core team members are OK with it. It can be a point of discussion for us separately.